syntax = "proto3";


service Segment {
  rpc seg (SegRequest) returns (SegResponse) {}
  rpc pos (SegRequest) returns (PosResponse) {}
  rpc add_word (AddWordRequest) returns (Bool) {}
  rpc delete_word (DeleteWordRequest) returns (Bool) {}
}


message SegRequest {
  string content = 1;
  string model = 2;
  bool enable_stop_word = 3;
  bool use_ner = 4;
}

message SegResponse {
  message Term {
    string word = 1;
    int32 start_index = 2;
    int32 end_index = 3;
  }
  repeated Term terms = 1;
}

message PosResponse {
  message Term {
    string word = 1;
    int32 start_index = 2;
    int32 end_index = 3;
    string pos = 4;
  }
  repeated Term terms = 1;
}

message Bool {
  bool status = 1;
}

message AddWordRequest {
  string word = 1;
  string pos = 2;
  int32 freq = 3;
}

message DeleteWordRequest {
  string word = 1;
}
